crates/nu-command/src/filesystem/glob.rs

```diff
@@ -138,6 +162,16 @@ impl Command for Glob {
                 },
             )
             .flatten()
+            .filter(|entry| {
+                let file_type = entry.file_type();
+
+                match () {
+                    () if no_dirs && file_type.is_dir() => false,
+                    () if no_files && file_type.is_file() => false,
+                    () if no_symlinks && file_type.is_symlink() => false,
+                    () => true,
```

Comment:
The only thing keeping me from landing this PR is this weird syntax of `match ()`. Is there no way to rewrite this more idiomatic and intuitive?

Reply:
The most intuitive way to write this, for me, _is_ a series of ifs. The match is there because it gets clippy out of the way from collapsing it into a boolean expression, however, I certainly wouldn't call it idiomatic. The filter can be rewritten into a terse boolean expression:

```rs
            .filter(|entry| {
                let file_type = entry.file_type();

                !(no_dirs && file_type.is_dir()
                    || no_files && file_type.is_file()
                    || no_symlinks && file_type.is_symlink())
            })
```

I can push this change and put this to rest if you want

Reply:
I'm fine with something other than that match that makes clippy happy.